magefiles/utils.go

```diff
@@ -154,3 +156,65 @@ func renderTemplate(destination, templatePath string, templateData interface{},
 
 	return nil
 }
+
+func cleanupQuay(quayService quay.QuayService, quayOrg string) error {
+
+	r, err := regexp.Compile(`^(e2e-demos|has-e2e)`)
+	if err != nil {
+		return err
+	}
+
+	repos, err := quayService.GetAllRepositories(quayOrg)
+	if err != nil {
+		return err
+	}
+
+	// Key is the repo name without slashes which is the same as robot name
+	// Value is the repo name with slashes
+	reposMap := make(map[string]string)
+
+	for _, repo := range repos {
+		if r.MatchString(repo.Name) {
+			sanitizedRepoName := strings.ReplaceAll(repo.Name, "/", "") // repo name without slashes
+			reposMap[sanitizedRepoName] = repo.Name
+		}
+	}
+
+	robots, err := quayService.GetAllRobotAccounts(quayOrg)
+	if err != nil {
+		return err
+	}
+
+	r, err = regexp.Compile(fmt.Sprintf(`^%s\+(e2e-demos|has-e2e)`, quayOrg))
+	if err != nil {
+		return err
+	}
+
+	const timeFormat = "Mon, 02 Jan 2006 15:04:05 -0700"
+
+	// Deletes robots and their repos with correct prefix if created more than 24 hours ago
+	for _, robot := range robots {
+		parsed, err := time.Parse(timeFormat, robot.Created)
+		if err != nil {
+			return err
+		}
+
+		// If robot.Name has correct prefix and was created more than 24 hours ago
+		if r.MatchString(robot.Name) && time.Since(parsed) > 24*time.Hour {
+			// Robot name without the name of org which is the same as previous sanitizedRepoName
+			// redhat-appstudio-qe+e2e-demos turns to e2e-demos
+			sanitizedRepoName := strings.Split(robot.Name, "+")[1]
+			if repo, exists := reposMap[sanitizedRepoName]; exists {
+				deleted, err := quayService.DeleteRepository(quayOrg, repo)
+				if !deleted || err != nil {
```

Comment:
for whatever reason there could be a case [when the repo is no longer present in registry](https://github.com/redhat-appstudio/image-controller/blob/7e9bc2f236ee690ea83a6e437e82cf54ce7b2128/pkg/quay/quay.go#L121) - I think that should be treated differently